### PR TITLE
[HomeKit] Added support to Xcode 14.1 b1-b3

### PR DESCRIPTION
**Pull request by SotoiGhost**
* Matter properties are still missing

### CODE REVIEW

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/homekit.cs [hunk 1]
```diff
@@ -1921,75 +1932,14 @@ interface HMAccessControl {
 
 	}
 
-	[iOS (15,2), Watch (8,3), TV (15,2), MacCatalyst (15,2)]
-	[BaseType (typeof (NSObject))]
-	[DisableDefaultCtor]
-	interface HMMatterHome : NSCopying, NSSecureCoding
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/homekit.cs:1927]: I don't think so, since it won't compile without them.

**Review by chamons — APPROVED** [review]
Approved assuming apidiff comes back clean now.

**Review by SotoiGhost — COMMENTED** [review]
LGTM! 👍🏼

**Review by rolfbjarne — CHANGES_REQUESTED** [review]
Looks mostly good, but the api diff is still red on tvOS, Mac Catalyst and watchOS:

```
Removed Type HomeKit.HMFetchRoomHandler
Removed Type HomeKit.HMMatterHome
Removed Type HomeKit.HMMatterRequestHandler
Removed Type HomeKit.HMMatterRoom
Removed Type HomeKit.HMMatterTopology
```

File: src/homekit.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tj-devel709) [reply src/homekit.cs:1927]: Was this addressed?

[the remark this answers is not in the document]
Comment (tj-devel709) [reply src/homekit.cs:1927]: Oh nevermind! I see that's what those added classes are there for!

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**